pennylane/transforms/dynamic_one_shot.py

```diff
@@ -284,8 +285,18 @@ def gather_mcm(measurement, samples):
         mcm_samples = np.concatenate(mcm_samples, axis=1)
         meas_tmp = measurement.__class__(wires=wires)
         return meas_tmp.process_samples(mcm_samples, wire_order=wires)
-    mcm_samples = np.array([mv.concretize(dct) for dct in samples]).reshape((-1, 1))
-    use_as_is = len(mv.measurements) == 1
+    if isinstance(measurement, ProbabilityMP):
+        if len(mv.measurements) > 1:
+            raise ValueError(
+                "Cannot use qml.probs() when measuring multiple mid-circuit measurements collected "
+                "using arithmetic operators. To collect probabilities for multiple mid-circuit "
+                "measurements, use a list of mid-circuit measurements with qml.probs()."
+            )
```

Comment:
This is also not supported with `defer_measurements`. The same error is raised in the `qml.probs` constructor function, so perhaps instead of raising it in both of those places, we can just put it in `ProbabilityMP.__init__`.